pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -325,6 +325,9 @@ func (gb *GraphBuilder) Run(stopCh <-chan struct{}) {
 	// Set up the stop channel.
 	gb.monitorLock.Lock()
 	gb.stopCh = stopCh
+	if stopCh == nil {
```

Comment:
Is this masking some problem we should know about?

Reply:
No, nil is valid as a stop channel if you don't care about stopping (blocks receives forever). This use required calling Run to result in a non nil stopCh, so ensure that happens internally

Reply:
> No, nil is valid as a stop channel if you don't care about stopping (blocks receives forever). This use required calling Run to result in a non nil stopCh, so ensure that happens internally

I'm fairly certain that codebase will panic on that.  All the wait methods receive from the channel unconditionally, like this: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go#L117

Why is this stopCh different than the other stopChs?

Reply:
> I'm fairly certain that codebase will panic on that

It does not. We were passing nil before #57340 and it didn't cause panics.

> All the wait methods receive from the channel unconditionally, like this

And that works just like a non-nil channel that is blocking receives. See https://play.golang.org/p/o6LVWLoSXQk

> Why is this stopCh different than the other stopChs?

Because this controller overloads the meaning of stopCh to also mean "non-nil means Run has been called". I can switch this to an `running bool` field if preferred.

Reply:
> I can switch this to a running bool field if preferred.

It is.  Receiving on a nil channel is unexpected.  I didn't even realize it was a thing that "worked".  We've used `wait.NeverStop` for never stop elsewhere.

Reply:
updated